rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,499 @@
+[2]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix
+[3]: https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix
+
+In general, the format for writing documentation strings is **not formally specified**.
+
+Among the formats encountered in the wild, the one used in `nixpkgs/lib` is the only one intended to be rendered as part of an API documentation, via the nixdoc third-party tool, whose syntax has not been standardized.
+
+### Impossible to differentiate from internal comments
+
+The lack of a formal definition of a doc-comment also means there is no reliable way to distinguish them from internal comments, which makes it impossible to access from doc-comments from tooling. Futhermore it makes it very hard to generate accurate and complete reference documentation.
+
+### References to the problems above
+
+This curated link collection highlights the Nix ecosystem's inconsistencies, a primary focus of this RFC.
+
+#### nixpkgs - comment examples
+
+- [lib/attrsets](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/lib/attrsets.nix)
+- [trivial-builders](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/pkgs/build-support/trivial-builders/default.nix)
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/5323fbf70331f8a7c47f1b4f49841cf74507f77f/nixos/lib/make-disk-image.nix)
+
+# Design
+[design]: #detailed-design
+
+In the following we give a comprehensive overview to our decision that we've made. Detailed arguments for and against every decision can be found in the [Decisions](#Decisions) section
+
+## CommonMark
+
+**CommonMark according to [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md) is the content of all doc-comments.**
+
+Adopting CommonMark as the content for all doc-comments brings the benefits of widely accepted and understood documentation format in tech projects, while maintaining profitability and consistency within the Nix ecosystem by aligning with existing [NixOS/RFC-72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md).
+
+## `/** */` is the doc-comment format
+
+The decision to use /** to start a doc-comment ensures a unique distinction from regular comments while still allowing seamless writing without IDE or editor support. A single choice provides the best developer experience and simplifies tooling.
+
+## Placement
+
+**The placement describes the relationship between doc-comments and the documentable node.**
+
+A **documentable node** can be:
+
+- Expression 
+- Binding 
+- Lambda Formal
+
+The following rules apply in descending order of precedence:
+
+- Doc-comments are placed before the **documentable node**. Only whitespace is allowed in between. ([Examples](#basic-examples)) 
```

Comment:
"Only white space" is a bit surprising, I would expect “white space or non-doc comments”.

That’s the rule Rust uses, and it is practically useful — oftentimes, you want to document both the public API with a doc comment, as well as private impl details with a non-doc comment.

Reply:
Also thought of this. Maybe we should indeed allow regular comments too. Thanks for the suggestion if rust uses this convention we should maybe follow the same.
I was concerned too, because in nixpkgs.lib there are many 
`# TODO: @user fix this sometimes later.`
The reason we put the rule there in the first place was because it visually interrupts the connection to the documentable node.